4_Zadania_dodatkowe/6.InfiniteSlider/js/app.js

```diff
@@ -0,0 +1,71 @@
+$(document).ready(function () {
+   // console.log('test');
+
+    var elementPrevButton = $('#prevPicture');
+    var elementNextButton = $('#nextPicture');
+    // console.log(elementPrevButton, elementNextButton);
+
+    var elementListLi = $('li');
+    // console.log(elementListLi);
+
+    var elementUl = $('ul');
+    // console.log(elementUl);
+
+    var visibleImageIndex = 1;
+    var imageWidth = 404;
+    // console.log(visibleImageIndex, imageWidth);
+
+    var firstElement = elementListLi.first().clone();
+    var lastElement = elementListLi.last().clone();
+    // console.log(firstElement);
+    // console.log(lastElement);
+    elementUl.append(firstElement);
+    elementUl.prepend(lastElement);
+
+
+    function moveRight() {
+        visibleImageIndex++;
+        elementUl.animate({
+                'right': imageWidth * visibleImageIndex,
+            },
+            {
+                complete: function () {
+                    if (visibleImageIndex > 6) {
```

Comment:
`6` - magical number
your script should work with any number of images, don't hardcode this value
read it from `elementListLi`

this applies to all other hardcoded numerical values in this file: `404`, `2828` (let browser do the math!)